apps/zbugs/shared/queries.ts

```diff
@@ -42,39 +42,63 @@ const issueRowSort = z.object({
 
 type IssueRowSort = z.infer<typeof issueRowSort>;
 
+const defineAuthQuery = defineQueryWithContextType<AuthData | undefined>();
+
 export const queries = {
-  allLabels: syncedQuery('allLabels', z.tuple([]), () => builder.label),
+  allLabels: defineQuery(
+    'allLabels',
+    {
+      validator: z.undefined(),
+    },
+    () => builder.label,
+  ),
 
-  allUsers: syncedQuery('allUsers', z.tuple([]), () => builder.user),
+  allUsers: defineQuery(
+    'allUsers',
+    {
+      validator: z.undefined(),
+    },
+    () => builder.user,
+  ),
 
-  allProjects: syncedQuery('allProjects', z.tuple([]), () => builder.project),
+  allProjects: defineQuery(
+    'allProjects',
+    {
+      validator: z.undefined(),
+    },
+    () => builder.project,
+  ),
 
-  user: syncedQuery('user', idValidator, userID =>
-    builder.user.where('id', userID).one(),
+  user: defineQuery(
+    'user',
+    {
+      validator: idValidator,
+    },
+    ({args: userID}) => builder.user.where('id', userID).one(),
   ),
 
-  labelsForProject: syncedQuery(
+  labelsForProject: defineQuery(
     'labels',
-    z.tuple([
-      z.object({
+    {
+      validator: z.object({
         projectName: z.string(),
       }),
-    ]),
-    ({projectName}: {projectName: string}) =>
+    },
+    ({args: {projectName}}) =>
       builder.label.whereExists('project', q =>
         q.where('lowerCaseName', projectName.toLocaleLowerCase()),
       ),
   ),
 
-  issuePreloadV2: syncedQueryWithContext(
+  issuePreloadV2: defineAuthQuery(
     'issuePreloadV2',
-    z.tuple([
-      z.object({
+    {
+      validator: z.object({
         userID: z.string(),
         projectName: z.string(),
       }),
-    ]),
-    (auth: AuthData | undefined, {userID, projectName}) =>
+    },
+    ({ctx: auth, args: {userID, projectName}}) =>
```

Comment:
In this file we should name this field `ctx` so it can be passed like `{ctx`. Perhaps this is a reason that the field should actually be called `context`.